Source/Engine/Tools/MaterialGenerator/MaterialGenerator.Textures.cpp

```diff
@@ -444,6 +444,48 @@ void MaterialGenerator::ProcessGroupTextures(Box* box, Node* node, Value& value)
         value = box == gradientBox ? gradient : distance;
         break;
     }
+    // World Triplanar Texture
+    case 16:
+    {
+        // Get input boxes
+        auto textureBox = node->GetBox(0);
+        auto scaleBox = node->GetBox(1);
+        
+        if (!textureBox->HasConnection())
+        {
+            // No texture to sample
+            value = Value::Zero;
+            break;
+        }
+
+        if (!CanUseSample(_treeType))
+        {
+            // Must sample texture in pixel shader
+            value = Value::Zero;
+            break;
+        }
+
+        const auto texture = eatBox(textureBox->GetParent<Node>(), textureBox->FirstConnection());
+        const auto scale = tryGetValue(scaleBox, node->Values[0]);
+
+        auto result = writeLocal(Value::InitForZero(ValueType::Float4), node);
+
+        const String triplanarTexture = String::Format(TEXT(
+            "   float3 worldPos = input.WorldPosition.xyz * {1} / 1000;\n"
+            "   float3 normal = input.TBN[2];\n"
+
+            "   {2} += {0}.Sample(SamplerLinearWrap, worldPos.yz) * abs(dot(normal, float3(1,0,0)));\n"
```

Comment:
Optionally: provide `float Blend = 1.0f` input to the node and apply it here to control the blending threshold:
```hlsl
{2} += {0}.Sample(SamplerLinearWrap, worldPos.yz) * pow(abs(dot(normal, float3(1,0,0)), {3}));
// {3} = Blend
```
(would be nice to have so the transition on the object edges could be bigger or smaller)

Reply:
Fixed.